contracts/extensions/WhitelistExtension.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.23;
+
+import { IOrderMixin } from "@1inch/limit-order-protocol-contract/contracts/interfaces/IOrderMixin.sol";
+import { ExtensionBase } from "../ExtensionBase.sol";
+
+/**
+ * @title Whitelist Extension
+ * @notice Abstract contract designed to check resolvers from orders in whitelist within the post-interaction phase of order execution.
+ * Ensures that only transactions from whitelisted resolvers are processed, enhancing security and compliance.
+ */
+abstract contract WhitelistExtension is ExtensionBase {
+    error ResolverIsNotWhitelisted();
+
+    /**
+     * @dev Validates whether the resolver is whitelisted.
+     * @param whitelist Whitelist is tighly packed struct of the following format:
+     * ```
+     * struct WhitelistDetails {
+     *     bytes4 auctionStartTime;
+     *     (bytes10,bytes2)[N] resolversAddressesAndTimeDeltas;
+     * }
+     * ```
+     * @param resolver The resolver to check.
+     * @return Whether the resolver is whitelisted.
+     */
+    function _isWhitelisted(bytes calldata whitelist, address resolver) internal view virtual returns (bool) {
+        unchecked {
+            uint256 allowedTime = uint32(bytes4(whitelist[0:4])); // initially set to auction start time
+            whitelist = whitelist[4:];
+            uint256 whitelistSize = whitelist.length / 12;
```

Comment:
If I get it correctly, we calculate this value in line 65 and can pass it to this function as a parameter.

> uint8 resolversLength = uint8(extraData[extraData.length - 1]) >> 3;